cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -258,9 +261,12 @@ public static KafkaCluster fromAssembly(StatefulSet ss, String namespace, String
      * @return List with generated ports
      */
     private List<ServicePort> getHeadlessServicePorts() {
-        List<ServicePort> ports = new ArrayList<>(2);
+        List<ServicePort> ports = new ArrayList<>(3);
         ports.add(createServicePort(CLIENT_PORT_NAME, CLIENT_PORT, CLIENT_PORT, "TCP"));
         ports.add(createServicePort(REPLICATION_PORT_NAME, REPLICATION_PORT, REPLICATION_PORT, "TCP"));
+        if (isMetricsEnabled()) {
+            ports.add(createServicePort(metricsPortName, metricsPort, metricsPort, "TCP"));
```

Comment:
I don't think we need the metrics port on all services ... either the regular one or the headless should be enough, but not both, or?